src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,67 @@
+        internal static DiagnosticDescriptor RuleDuplicatedValue = new DiagnosticDescriptor(RuleId,
+                                                                                            s_localizableTitle,
+                                                                                            s_localizableMessageRuleDuplicatedValue,
+                                                                                            DiagnosticCategory.Design,
+                                                                                            DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                            isEnabledByDefault: false,
+                                                                                            helpLinkUri: null,
+                                                                                            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private static readonly LocalizableString s_localizableMessageRuleDuplicatedBitwiseValuePart = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.EnumShouldNotHaveDuplicatedValuesMessageDuplicatedBitwiseValuePart), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        internal static DiagnosticDescriptor RuleDuplicatedBitwiseValuePart = new DiagnosticDescriptor(RuleId,
+                                                                                                   s_localizableTitle,
+                                                                                                   s_localizableMessageRuleDuplicatedBitwiseValuePart,
+                                                                                                   DiagnosticCategory.Design,
+                                                                                                   DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                                   isEnabledByDefault: false,
+                                                                                                   helpLinkUri: null,
+                                                                                                   customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolAction(symbolContext =>
```

Comment:
I couldn't find any good way to achieve something similar using operations. I hope that's fine like this.

Reply:
I think we can write this as a much simpler language agnostic analyzer, completely based on operations as follows:

1. Register SymbolStartAction for NamedTypeSymbols.
2. Bail out from callback if type is not enum
3. Register nested operation callback for `OperationKind.FieldInitializer`.
4. Bail out in operation callback if:
   1. InitializedFields.Count != 1
   2. Operation does not have constant value
   3. Has any child IFieldReferenceOperation whose underlying field has same containing type
5. Otherwise, add to a dictionary of constant value to list of fields with that value
6. In SymbolEndAction, report diagnostics for all dictionary entries that have more than one field in the value - one diagnostic for each field excluding the one that is lexically the first field.
3.

Reply:
Yay that does work really well! I'll push the changes tomorrow.

Reply:
Regarding point 4. iii, I think the condition shall be more complex otherwise we won't report for duplicated bitwise parts nor for binary expression using field ref like `Value3 = Value2 * 42` (no idea if someone writes that but that's allowed by the syntax).

Reply:
Also we need to cover the bitwise duplication `Value1 | Value1` but we need to be careful about other binary expressions `Value1 | Value2 + Value1` shouldn't be marked with the bitwise error.

Reply:
There is also an issue with uninitialized values 
```csharp
public enum MyEnum
{
    Value1 = 1,
    Value2,
    Value3 = 2,
}
```
I can't handle this case with the new technique.